actions/get_dependencies/get_dependencies.sh

```diff
@@ -279,7 +279,7 @@ for depend in "$@"; do
     get_graphviz
   elif [ "${depend}" = "lapacke" ]; then
     get_lapacke
-  elif [ "${depend}" = "libint" ]; then
```

Comment:
This is going to get messy quickly as we start cacheing more dependencies. It would be better to only invoke this script if we want to build the dependency. Basically handle whether we're building each dependency up a level.

Reply:
Hi Ryan. I update the pull request. I create another workflow for handling the caches before getting dependencies. For the get_dependencies.sh, I move the check condition of caching libint onto the get_libint() function.